### PR TITLE
Add Crossgen pack for crossgen2 tool

### PR DESCRIPTION
**Pull request by dagood**
For https://github.com/dotnet/runtime/issues/906.

~Requires support in the shared framework tooling: PR https://github.com/dotnet/arcade/pull/4650.~

~I have this test build that produced the Crossgen pack by temporarily inlining the sharedfx tooling update: https://dev.azure.com/dnceng/internal/_build/results?buildId=487241~

New test build for Crossgen2 pack: https://dev.azure.com/dnceng/internal/_build/results?buildId=488118&view=results

### CODE REVIEW

File: src/installer/pkg/projects/netcoreapp/pkg/crossgenRIDs.props [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  <PropertyGroup>
+    <BuildOnUnknownPlatforms>false</BuildOnUnknownPlatforms>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <PackageRID Condition="'$(PackageRID)' == ''">$(OutputRid)</PackageRID>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <OfficialBuildRID Include="linux-x64" />
```
Comment (dagood) [inline src/installer/pkg/projects/netcoreapp/pkg/crossgenRIDs.props]: This produced a linux-x64 crossgen pack in my local builds, but I don't see one in the mock official build I ran. Looking into it.

**Comment by dagood** [comment]
Filed tracking issues:
* https://github.com/dotnet/runtime/issues/1869 "Stop producing Crossgen2 transport package (use Crossgen2 pack downstream instead)"
* https://github.com/dotnet/runtime/issues/1867 "Integrate Crossgen2 pack infrastructure into the shared framework tooling in Arcade"

**Comment by dagood** [comment]
This is good to go, PTAL.

Here are some of the nupkgs from the test build (links need Microsoft auth)

* [Microsoft.NETCore.App.Crossgen2.win-x64](https://dev.azure.com/dnceng/_apis/resources/Containers/3242233/IntermediateUnsignedArtifacts?itemPath=IntermediateUnsignedArtifacts%2FWindows_NT_x64%2FShipping%2FMicrosoft.NETCore.App.Crossgen2.win-x64.5.0.0-alpha.1.20067.5.nupkg)
* [Microsoft.NETCore.App.Crossgen2.linux-x64](https://dev.azure.com/dnceng/_apis/resources/Containers/3242233/IntermediateUnsignedArtifacts?itemPath=IntermediateUnsignedArtifacts%2FLinux_x64%2FShipping%2FMicrosoft.NETCore.App.Crossgen2.linux-x64.5.0.0-alpha.1.20067.5.nupkg)
* [Microsoft.NETCore.App.Crossgen2.linux-musl-x64](https://dev.azure.com/dnceng/_apis/resources/Containers/3242233/IntermediateUnsignedArtifacts?itemPath=IntermediateUnsignedArtifacts%2FLinux_musl_x64%2FShipping%2FMicrosoft.NETCore.App.Crossgen2.linux-musl-x64.5.0.0-alpha.1.20067.5.nupkg)

**Review by fadimounir — APPROVED** [review]
LGTM, but I'd like someone else more familiar with the build system to also sign off on the chagnes

**Comment by fadimounir** [comment]
Thanks for doing this work @dagood !

**Comment by dagood** [comment]
Test failures on `runtime (CoreCLR Pri0 Test Run Windows_NT x64 checked)` are a persistent known issue: https://github.com/dotnet/coreclr/issues/26241 (https://github.com/dotnet/runtime/issues/702). I'll disregard and go ahead with the merge if approved.

**Review by MichaelSimons — APPROVED** [review]

**Pull request merged**